owasp-top10-2021-apps/a8/amarelo-designs/app/app.py

```diff
@@ -19,12 +23,9 @@ def login():
     
         if username == "admin" and password == "admin":
             token = str(uuid.uuid4().hex)
-            cookie = { "username":username, "admin":True, "sessionId":token }
-            pickle_resultado = pickle.dumps(cookie)
-            encodedSessionCookie = base64.b64encode(pickle_resultado)
-            resp = make_response(redirect("/user"))
-            resp.set_cookie("sessionId", encodedSessionCookie)
-            return resp
+            admin_token.append(token)
```

Comment:
Admin tokens are stored in a local variable. But are they removed when expired?
What would happen if we have a HA setup with 2 instances of this app? Would an user authenticated as admin in the first one be recognized as such in the second one?

Reply:
It was fixed with:
mariadb + event scheduler + logging.

<img width="1046" alt="Captura de Tela 2022-04-14 às 13 53 37" src="https://user-images.githubusercontent.com/28883132/163436830-695e50de-f7f8-4011-ad70-6711b1f2d50a.png">